From trenn@suse.de  Tue Feb  3 14:51:29 2009
From: Venkatesh Pallipadi <venkatesh.pallipadi@intel.com>
Date: Tue, 27 Jan 2009 17:38:43 +0100
Subject: cpuidle: update the last_state acpi cpuidle reflecting actual state entered
To: stable@kernel.org
Cc: Len Brown <len.brown@intel.com>, Venkatesh Pallipadi <venkatesh.pallipadi@intel.com>
Message-ID: <1233074343-23414-2-git-send-email-trenn@suse.de>

From: Venkatesh Pallipadi <venkatesh.pallipadi@intel.com>

commit addbad46ed0906cd584784423b9d0babc7476446

reflect the actual state entered in dev->last_state, when actaul state entered
is different from intended one.

Signed-off-by: Venkatesh Pallipadi <venkatesh.pallipadi@intel.com>
Signed-off-by: Len Brown <len.brown@intel.com>
Cc: Thomas Renninger <trenn@suse.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/acpi/processor_idle.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/acpi/processor_idle.c
+++ b/drivers/acpi/processor_idle.c
@@ -1587,6 +1587,7 @@ static int acpi_idle_enter_bm(struct cpu
 
 	if (acpi_idle_bm_check()) {
 		if (dev->safe_state) {
+			dev->last_state = dev->safe_state;
 			return dev->safe_state->enter(dev, dev->safe_state);
 		} else {
 			local_irq_disable();
